contrib/storage-ipfs/src/main/java/org/apache/drill/exec/store/ipfs/IPFSStoragePluginConfig.java

```diff
@@ -0,0 +1,191 @@
+
+        @JsonCreator
+        public static IPFSTimeOut of(String which) {
+            switch (which) {
+                case "find-provider":
+                    return FIND_PROV;
+                case "find-peer-info":
+                    return FIND_PEER_INFO;
+                case "fetch-data":
+                    return FETCH_DATA;
+                default:
+                    throw new InvalidParameterException("Unknown key for IPFS timeout config entry: " + which);
+            }
+        }
+
+        @Override
+        public String toString() {
+            return this.which;
+        }
+    }
+
+    @JsonProperty("groupscan-worker-threads")
+    private final int numWorkerThreads;
+
+    @JsonProperty
+    private final Map<String, FormatPluginConfig> formats;
+
+    @JsonCreator
+    public IPFSStoragePluginConfig(
+        @JsonProperty("host") String host,
+        @JsonProperty("port") int port,
+        @JsonProperty("max-nodes-per-leaf") int maxNodesPerLeaf,
+        @JsonProperty("ipfs-timeouts") Map<IPFSTimeOut, Integer> ipfsTimeouts,
+        @JsonProperty("groupscan-worker-threads") int numWorkerThreads,
+        @JsonProperty("formats") Map<String, FormatPluginConfig> formats) {
+        this.host = host;
+        this.port = port;
+        this.maxNodesPerLeaf = maxNodesPerLeaf > 0 ? maxNodesPerLeaf : 1;
+        //TODO Jackson failed to deserialize the ipfsTimeouts map causing NPE
+        if (ipfsTimeouts != null) {
+            ipfsTimeoutDefaults.forEach(ipfsTimeouts::putIfAbsent);
+        } else {
+            ipfsTimeouts = ipfsTimeoutDefaults;
+        }
+        this.ipfsTimeouts = ipfsTimeouts;
+        this.numWorkerThreads = numWorkerThreads > 0 ? numWorkerThreads : 1;
+        this.formats = formats;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public int getMaxNodesPerLeaf() {
+        return maxNodesPerLeaf;
+    }
+
+    public int getIpfsTimeout(IPFSTimeOut which) {
+        return ipfsTimeouts.get(which);
+    }
+
+    public Map<IPFSTimeOut, Integer> getIpfsTimeouts() {
+        return ipfsTimeouts;
+    }
+
+    public int getNumWorkerThreads() {
+        return numWorkerThreads;
+    }
+
+    public Map<String, FormatPluginConfig> getFormats() {
+        return formats;
+    }
+
+    @Override
+    public int hashCode() {
+        String host_port = String.format("%s:%d[%d,%s]", host, port, maxNodesPerLeaf, ipfsTimeouts);
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((host_port == null) ? 0 : host_port.hashCode());
+        result = prime * result + ((formats == null) ? 0 : formats.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
```

Comment:
This can be consolidate in single above `if` with `if (obj == null || getClass() != ...)`

Reply:
Changed in 282a89d.